ambry-rest/src/main/java/com.github.ambry.rest/NettyMessageProcessor.java

```diff
@@ -117,8 +120,10 @@ public void channelInactive(ChannelHandlerContext ctx) {
     logger.trace("Channel {} inactive", ctx.channel());
     nettyMetrics.channelDestructionRate.mark();
     if (request != null && request.isOpen()) {
-      logger.error("Request {} was aborted because the channel became inactive", request.getUri());
-      responseChannel.onResponseComplete(new ClosedChannelException());
+      logger.error("Request {} was aborted because the channel {} became inactive", request.getUri(), ctx.channel());
+      onRequestAborted(new ClosedChannelException());
+    } else {
+      close();
```

Comment:
As per the documentation, this method gets called only once and *after* the channel is closed. So remind me why you have to call close here? Perhaps you could update the documentation as well.